patch-core.js

```diff
@@ -35,15 +58,35 @@ if (!https.request[patchMarker]) {
  * Ref: https://github.com/nodejs/node/commit/5118f31
  */
 https.get = function (_url, _options, cb) {
-    let options;
-    if (typeof _url === 'string' && _options && typeof _options !== 'function') {
-      options = Object.assign({}, url.parse(_url), _options);
-    } else if (!_options && !cb) {
-      options = _url;
-    } else if (!cb) {
-      options = _url;
-      cb = _options;
+  // Support for both
+  // https.get(options[, callback])
+  // https.get(url[, options][, callback])
+  if (typeof _url !== 'string' && !cb) {
+    cb = _options;
+    _options = _url;
+    _url = undefined;
+  }
+  if (!_url && _options) {
+    _url = url.format(_options);
+  }
+  var options;
+  if (typeof _options === 'string') {
+    options = url.parse(_options);
+  } else {
+    options = url.parse(_url);
+    if (_options) {
+      if (_options && Object.getOwnPropertySymbols && Object.getOwnPropertySymbols(_options)[0]) {
```

Comment:
Again, this logic now needs more abstraction and explaining :)

Reply:
I did abstract this into a re-usable function! Please check the update!